### PR TITLE
Fx: Remove unused converters

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

We had a bug in `main` before production because a sns aggregator converter that wasn't used was creating problems.

The issue with the converter has been fixed. Yet, we don't need to do some conversions to store the data in snsQueryStore.

In this PR, I removed the fields that the aggregator exposes, but we don't use for the snsQueryStore.

# Changes

* Remove fields `swap_params`, `init` and `lifecycle` from CachedSns type and converter. These fields are not used to store data in snsQueryStore.

# Tests

* Remove a test case that was checking a removed field.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: frontend/src/tests/mocks/sns-aggregator.mock.ts [hunk 1]
```diff
@@ -19,279 +19,6 @@ export const aggregatorSnsMockDto: CachedSnsDto = {
 };
 
 // It should match the converted response from sns-aggregator.mock.json with the same `index` value
-export const aggregatorSnsMock: CachedSns = {
```
Comment (lmuntaner) [inline frontend/src/tests/mocks/sns-aggregator.mock.ts:22]: Moved to `sns-aggregator-converters.spec.ts`

File: frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts [hunk 2]
```diff
@@ -3,39 +3,248 @@ import {
...
-        aggregatorSnsMock,
-      ]);
-    });
-
-    it("converts aggregator types to ic-js types if no swap params", () => {
-      const aggregatorDto: CachedSnsDto = {
-        ...aggregatorSnsMockDto,
-        swap_params: {
-          params: null,
+  const mockData: CachedSnsDto = {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:10]: This was created in the second "describe" below. But now I used it in both; therefore, I moved it up.

File: frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts [hunk 3]
```diff
@@ -3,39 +3,248 @@ import {
...
+    lifecycle: {
+      decentralization_sale_open_timestamp_seconds: 1690786778,
+      lifecycle: 2,
+    },
+  };
 
-  describe("convertDtoToSnsSummary", () => {
-    const mockData: CachedSnsDto = {
+  describe("convertDtoData", () => {
+    const mockConvertedData: CachedSns = {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:247]: This is the object that was in `sns-aggregator.mock.ts`

File: frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts [hunk 4]
```diff
@@ -3,39 +3,248 @@ import {
...
 
 describe("sns aggregator converters utils", () => {
-  describe("convertDtoData", () => {
-    it("converts aggregator types to ic-js types", () => {
-      expect(convertDtoData([aggregatorSnsMockDto])).toEqual([
-        aggregatorSnsMock,
-      ]);
-    });
-
-    it("converts aggregator types to ic-js types if no swap params", () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:20]: This test has been removed.

**Comment by lmuntaner** [comment]
@dskloetd please review

Sorry, but the changes in the test do not look very clear, but moving big objects is always messy. Even in smaller PRs.

I tried to explain the changes and I hope I made it clearer.

File: frontend/src/lib/utils/sns-aggregator-converters.utils.ts [hunk 5]
```diff
@@ -327,15 +311,6 @@ const convertSnsData = ({
...
   derived_state: convertDerivedToResponse(derived_state),
-  swap_params: {
-    params: isNullish(swap_params.params)
-      ? []
-      : [convertSwapParams(swap_params.params)],
-  },
-  init: {
-    init: convertSwapInitParams(init.init),
-  },
-  lifecycle: convertLifecycleResponse(lifecycle),
```
Comment (dskloetd) [inline frontend/src/lib/utils/sns-aggregator-converters.utils.ts:338]: Do I see correctly that this basically reverts (part of?) https://github.com/dfinity/nns-dapp/pull/3024?

File: frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:247]: correct, no changes were done here.

Comment (lmuntaner) [reply frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:247]: ah, no wait! Yes, I removed the fields swap_params, init and lifecycle as I mentioned in the description.

Shall I revert to where it was before?

File: frontend/src/lib/utils/sns-aggregator-converters.utils.ts (hunk 5 above)
Comment (lmuntaner) [reply frontend/src/lib/utils/sns-aggregator-converters.utils.ts:338]: that's correct

File: frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts (hunk 3 above)
Comment (dskloetd) [reply frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:247]: Yes please. The changes should be visible from the diff so moving and changing shouldn't happen in the same PR.

Comment (lmuntaner) [reply frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts]: Done

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

File: frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts (hunk 2 above)
Comment (dskloetd) [reply frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:13]: I don't see it being used in the `describe` from lines 249-255. What do you mean exactly?

Comment (lmuntaner) [reply frontend/src/tests/lib/utils/sns-aggregator-converters.utils.spec.ts:13]: ah, true, I reverted it and I use the mock back again to make this PR clearer. Let me move this also.

**Review by dskloetd — APPROVED** [review]

**Pull request merged**